### ISSUE TITLE
[TASK] AverageX Contract

### ISSUE THREAD

**Issue by tokdaniel** [superfluid-finance/superfluid-stories#476]
## Overview
We need a contract, with the responsibility of producing TOREX contracts, and distributing AVG tokens.

**What?**
SuperApp & TOREX Factory

**How?**
The AverageX contract will be a factory of TOREX-es, and having control over the AVG token supply / and distribution

**Why?**

## Acceptance criteria
 - [x]  should provide an interface to to deploy TOREX-es.
 - [ ]  should take care of the AVG token rewards distribution
 - [x]  should provide a query interface for TOREX-s

## Links/Resources
 * ...

## Readiness criteria
- [x] Descriptive title
- [x] What?
- [ ] Why?
- [x] How?
- [x] Acceptance criteria complying with Definition of Done
- [ ] Estimate/Story points
- [x] Deadline: If not a part of an epic
- [x] Priority: Reflected as order on the bord, walking the board from the top and right
- [x] Assignee
- [x] Labels
- [x] Link to epic: If belonging to an epic
- [x] A task is always a separate ticket/issue
- [x] UI/UX Design (optional)
- [x] Blockers/Prerequisites (optional): Use Zenhub blocked by/blocking feature
